deploy/utils/dynamo_deployment.py

```
@@ -328,8 +335,10 @@ async def _detect_terminal_pod_failure(self) -> Optional[str]:
                 f"nvidia.com/dynamo-component={original_name}"
             )
             try:
-                pods = await self.core_api.list_namespaced_pod(
-                    namespace=self.namespace, label_selector=label_selector
+                pods = await retry_kubernetes_read(
+                    lambda: self.core_api.list_namespaced_pod(
+                        namespace=self.namespace, label_selector=label_selector
```

Ruff (0.16.1)

[warning] Function definition does not bind loop variable `label_selector`

(B023)